src/com/facebook/buck/go/GoDescriptors.java

```diff
@@ -142,6 +153,29 @@ static GoCompile createGoCompileRule(
         platform);
   }
 
+  @VisibleForTesting
+  static ImmutableMap<Path, Path> getPackageImportMap(
+      Path basePackageName, Iterable<Path> packageNameIter) {
+    Map<Path, Path> importMapBuilder = Maps.newHashMap();
+    ImmutableSortedSet<Path> packageNames = ImmutableSortedSet.copyOf(packageNameIter);
+
+    Path prefix = Paths.get("");
+    for (Path component : FluentIterable.of(new Path[]{Paths.get("")}).append(basePackageName)) {
+      prefix = prefix.resolve(component);
+
+      Path vendorPrefix = prefix.resolve("vendor");
+      for (Path vendoredPackage : packageNames.tailSet(vendorPrefix)) {
+        if (!vendoredPackage.startsWith(vendorPrefix)) {
+          continue;
```

Comment:
I get that part, but that's the outer loop right? The inner loop is comparing a candidate, for example `a/vendor`, with all the paths.

I'm saying that if you start alphabetically by using `tailSet` you can also stop alphabetically by using `break`:

```
a/b
a/b/c
a/vendor <= tailSet makes you start here
a/vendor/t
a/vendor/u
a/x <= break makes you stop here
a/y
a/z
```